Source/Foundation/bsfCore/Mesh/BsMesh.cpp

```diff
@@ -324,7 +324,7 @@ namespace bs
 
 		mIndexBuffer = IndexBuffer::create(ibDesc, mDeviceMask);
 
-		mVertexData = SPtr<VertexData>(bs_new<VertexData>());
+		mVertexData = bs_shared_ptr<VertexData>(bs_new<VertexData>());
```

Comment:
Could you elaborate on the benefit here?

Reply:
The code at this point is simply incorrect:
If at any point in time `GenAlloc` allocator begins to do clever things, the code 'will' break.
When mVertexData is released/freed currently it simply calls delete on the stored pointer, which was allocated by bs_new ( and thus using GenAlloc), and thus needs to be freed using  `bs_delete`